ynr/apps/elections/views.py

```diff
@@ -977,6 +977,9 @@ def form_valid(self, form):
             if lock:
                 suffix = "-lock"
                 pp = "Locked"
+                # If we're locking this, then the suggested posts
+                # can be deleted
+                pee.suggestedpostlock_set.all().delete()
```

Comment:
Can we wrap this function in a transaction so that if something throws an exception later in the function we roll this `delete()` back

Reply:
It's already in one, https://github.com/DemocracyClub/yournextrepresentative/blob/80930288262788740c681556516767dc93fdf1de/ynr/apps/elections/views.py#L966-L968